content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,140 @@
+Not all CNIs provide encryption in transit. If the chosen CNI lacks this
+feature, an alternative solution could be to use a service mesh to provide that
+functionality.
+
+The `etcd` datastore of the control plane should have controls to limit access.
+Furthermore, mutual TLS (mTLS) should be used to communicate securely with it.
+The certificate authority for this should be unique to `etcd`.
+
+## Pod security
+
+RBAC authorization is crucial but [cannot be specific enough on the Pods
+resource](/docs/concepts/security/rbac-good-practices/#workload-creation) (or
+on any resource that manages Pods). The only granularity is the API verbs on the
+resource itself, for example `create` on Pods. Indeed, without additional
+admission, the authorization to create these resources allows direct
+unrestricted access to on the schedulable nodes of a cluster.
+
+[Pod Security Admission](/docs/concepts/security/pod-security-admission/) is an
+admission controller to enable fine-grained authorization of Pod creation and
+updates. It replaces the deprecated (as v1.21) [Pod Security
+Policy](/docs/concepts/policy/pod-security-policy/) admission controller. The
+[Pod Security Standard](/docs/concepts/security/pod-security-standards/) dive
+deeper into the fields that can be restricted via the predefined profiles. For
+a hands-on tutorial on Pod Security, see the blog post "[Kubernetes 1.23: Pod
+Security Graduates to Beta](/blog/2021/12/09/pod-security-admission-beta/)".
+
+{{< note >}}
+[PodSecurityPolicy](http://localhost:1313/docs/concepts/policy/pod-security-policy/)
+will be removed in version 1.25.
+{{< /note >}}
+
+
+## Pod placement
+
+Pods that could be considered different tiers of sensitivity, for example an
+application pod and the Kubernetes API server, should be deployed onto separate
+nodes. This separation should be enforced to prevent pods accidentally being
+deployed onto the same node. This could be enforced with the following features:
+
+- [Node Selectors](/docs/concepts/scheduling-eviction/assign-pod-node/) are
+  key-value pairs as part of the pod specification that specifies which nodes
+  to deploy onto. These can be enforced at the namespace and cluster level with
+  the
+  [PodNodeSelector](/docs/reference/access-authn-authz/admission-controllers/#podnodeselector)
+  admission controller.
+- [PodTolerationRestriction](/docs/reference/access-authn-authz/admission-controllers/#podtolerationrestriction)
+  is an admission controller that allows administrators to add permitted
+  [tolerations](/docs/concepts/scheduling-eviction/taint-and-toleration/)
+  within a namespace. Pods within that namespace may only utilise those
+  specified in the namespace.
+
+## Secrets
+
+- Secrets required for pods should be stored within Kubernetes secrets as
+  opposed to alternatives such as ConfigMap.
+- Pods with secrets should have these automatically mounted through volumes.
+  This would allow adding secrets into the pod as environment variables or
```

Comment:
Pods with secrets should have these automatically mounted through volumes during the start of the pod.
This would allow adding secrets into the pod as files on the filesystem.

Reply:
I see you remove the part about env variables, do you want maybe to emphasise that env variables should be avoided for secrets, because of stack trace env dumps etc.? I think this would be also a nice note/warning for the [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) doc page?

Reply:
Yes, it would make sense to mention here, that env variables should not be used for secrets